doc/src/extras.rst

```diff
@@ -141,6 +141,383 @@ Logging cursor
+        replication cursor.  A reference to the cursor is provided in the
+        `ReplicationMessage` as an attribute.
+
+        .. warning::
+
+            When using replication with slots, failure to properly notify the
+            server by constantly consuming and reporting success at
+            appropriate times can eventually lead to "disk full" condition on
+            the server, because the server retains all the WAL segments that
+            might be needed to stream the changes via all of the currently
+            open replication slots.
+
+            On the other hand, it is not recommended to send a confirmation
+            after every processed message, since that will put an unnecessary
+            load on network and the server.  A possible strategy is to confirm
+            after every COMMIT message.
+
+    .. method:: stop_replication()
+
+        This method can be called on synchronous connections from the
+        ``consume()`` method of a ``consumer`` object in order to break out of
+        the endless loop in `consume_replication_stream()`.  If called on
+        asynchronous connection or outside of the consume loop, this method
+        raises an error.
+
+    .. method:: send_replication_feedback(write_lsn=0, flush_lsn=0, apply_lsn=0, reply=False)
+
+        :param write_lsn: a LSN position up to which the client has written the data locally
+        :param flush_lsn: a LSN position up to which the client has stored the
+                          data reliably (the server is allowed to discard all
+                          and every data that predates this LSN)
+        :param apply_lsn: a LSN position up to which the warm standby server
+                          has applied the changes (physical replication
+                          master-slave protocol only)
+        :param reply: request the server to send back a keepalive message immediately
+
+        Use this method to report to the server that all messages up to a
+        certain LSN position have been stored on the client and may be
+        discarded on the server.
+
+        This method can also be called with all default parameters' values to
+        just send a keepalive message to the server.
+
+        If the feedback message could not be sent, updates the passed LSN
+        positions in the cursor for a later call to
+        `flush_replication_feedback()` and returns `!False`, otherwise returns
+        `!True`.
+
+    .. method:: flush_replication_feedback(reply=False)
+
+        :param reply: request the server to send back a keepalive message immediately
+
+        This method tries to flush the latest replication feedback message
+        that `send_replication_feedback()` was trying to send but couldn't.
+
+        If *reply* is `!True` sends a keepalive message in either case.
+
+        Returns `!True` if the feedback message was sent successfully,
+        `!False` otherwise.
```

Comment:
`!True`? `!False`? What?

Also, should IMO return `True` if called without `reply` set and there are no queued feedback messages. It successfully sent the zero messages that were queued, that's not a failure.